src/mame/drivers/zr107.cpp

```diff
@@ -625,8 +625,8 @@ static INPUT_PORTS_START( midnrun )
 	PORT_DIPSETTING( 0x04, "3" )
 	PORT_DIPSETTING( 0x00, "4" )
 	PORT_DIPNAME( 0x02, 0x02, "Transmission Type" ) PORT_DIPLOCATION("SW:2")
-	PORT_DIPSETTING( 0x02, "Button" )
-	PORT_DIPSETTING( 0x00, "'T'Gate" ) //unused
+	PORT_DIPSETTING( 0x02, "'T'Gate" )
+	PORT_DIPSETTING( 0x00, "'Button" ) //unused
```

Comment:
Is it the “T'Gate” or “Button” type that’s unused, though?  Does the game expect sequential shift buttons/paddles, or a T gate shifter?  The actual unused setting should be commented.

(That aside, what is a “T'Gate”?  I mean, H gates are pretty common, but where would the gears be on a T gate?)

Reply:
It's the Button type that is unused.  The 'T'Gate is a gear selector that is moved left for auto transmission mode or can be pushed up/down to go up and down through the gears when in manual transmission mode.  The 'T'Gate gear selector can be seen clearly on this video: https://www.youtube.com/watch?v=da24oNpXcSA

Reply:
I commented this as unused but after looking at more cab pics, it seems to be dependent on the cabinet type. Deluxe and any EU cab set AT/MT as a T'Gate (hold the button to set transmission). Sitdown non-deluxe U.S. and JP cabs set AT/MT as a button. The AT/MT switch shouldn't be toggle anymore (my bad).

DX cab control panel: https://flyers.arcade-museum.com/?page=flyer&db=videodb&id=5620&image=2

EU upright cab control panel: https://www.youtube.com/watch?v=da24oNpXcSA

U.S. SD cab: https://flyers.arcade-museum.com/?page=flyer&db=videodb&id=1366&image=2 (huh. Konami decided on an up/down shifter with turbo button style).

JP SD twin cab: http://adb.arcadeitalia.net/media/mame.current/cpanels/windheat.png?release=202

Reply:
According to the Winding Heat operating manual available on archive.org, the dip switch for transmission is always OFF for a 'T'Gate transmission which makes sense as the manual contains is a diagram showing a 'T'Gate type selector.  Like you say, when it is set to 'T'Gate, the auto transmission signal probably requires to be continuously held.  But when it is set to 'Button', the auto transmission signal needs to be a toggle.  Currently MAME isn't working that way.  With the auto transmission dip switch set to OFF, the key assigned to the AT/MT works as a toggle (single press) so this would be correct behaviour for 'Button'.  With the auto transmission dip switch set to ON, the key assigned to the AT/MT works still works as a toggle but require 2 key presses to toggle which doesn't seem correct for 'T'Gate or 'Button'.